internal/parser/parser.go

```diff
@@ -972,15 +972,21 @@ func (p *parser) parsePropertyList() []*ast.Property {
 	var params []*ast.Property
 	for p.more() {
 		var param *ast.Property
-		switch _, tok, _ := p.peek(); tok {
+		switch pos, tok, lit := p.peek(); tok {
 		case token.IDENT:
 			param = p.parseIdentProperty()
 		case token.STRING:
 			param = p.parseStringProperty()
 		default:
 			// TODO(jsternberg): BadExpression.
+			p.errs = append(p.errs, ast.Error{
+				Msg: fmt.Sprintf("in property list, expected %s or %s, got %s (%q) at %s",
```

Comment:
Can you modify the message for this?

```
fmt.Sprintf("unexpected token for property key: %s (%q)", tok, lit)
```

It might also be good to encapsulate this in its own function where you consume the bad token and then consume any colon and expression that follows so it will catch things like `{5: a}`. This is with a special exception that if it sees a comma that it construct a `missing property` message (or something like that) and if it sees a colon it will construct a `missing property key` message. That should cover our bases.